src/Datadog.Trace/Agent/Jaeger/JaegerOptions.cs

```diff
@@ -7,6 +9,22 @@ public class JaegerOptions
     {
         internal const int DefaultMaxPayloadSizeInBytes = 4096;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JaegerOptions"/> class.
+        /// </summary>
+        /// <param name="settings"><see cref="TracerSettings"/> to read configuration from</param>
+        public JaegerOptions(TracerSettings settings = null)
```

Comment:
1. This is more like "conversion" than constructing a valid object.
I suggest changing it to:
  ```
  public static  JaegerOptions FromTracerSettings(TracerSettings settings)
  ```

2. However to be honest I do not see any reason to make such stuff public. What was wrong with having `private static JaegerOptions CreateJaegerOptions` in `Tracer`? It was used only there so I do not see any reason to extract it.

Reply:
You are right in this case. I did this mostly to illustrate how I see creating new setting from now on. Jaeger one uses only the settings that are already extracted from IConfigurationSource by TracerSettings, but in other cases we would probably want to add new configuration keys, then we would use the passed settings object to get to the IConfigurationSource instance nad extract what we want (`var myNewKeyValue = settings.ConfigurationSource.GetString("MY_NEW_KEY");`). I would rather not see such a code in a static Tracer method, that's why I moved it to JaegerOptions.

Reply:
>  we would probably want to add new configuration keys

Maybe we should do it? See: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#otlp-exporter

Reply:
I like the `static JaegerOptions FromTracerSettings(TracerSettings settings)` suggestion. It allows further separation of the logic for mapping settings to options which can make the test code involving options easier to manage. It also gets rid of the need for having the optional settings parameter (or multiple .ctors) so that test code doesn't also need to define settings code.

Reply:
resolve